src/pages/Governance/Stake/Index.tsx

```diff
@@ -38,68 +43,133 @@ export function StakePage() {
+    if (!stakingAmount) return;
+    await submitStake(parseInt(stakingAmount), operatorAddr, voterAddr);
   };
 
+  const onCloseErrorAlert = () => {
+    clearTransactionResponse();
+  };
+
+  useEffect(() => {
+    if (transactionResponse?.succeeded) {
+      setStakingAmount("");
+      setOperatorAddr("");
+      setVoterAddr("");
+    }
+  }, [transactionResponse]);
+
+  const voteOnSuccessSnackbarAction = (
+    <IconButton
+      size="small"
+      aria-label="close"
+      color="inherit"
+      onClick={onCloseErrorAlert}
+    >
+      <CloseIcon fontSize="small" />
+    </IconButton>
+  );
+
+  const voteOnSuccessSnackbar = transactionResponse !== null && (
+    <Snackbar
+      open={transactionResponse.succeeded}
+      anchorOrigin={{
+        vertical: "top",
+        horizontal: "center",
+      }}
+    >
+      <Alert
+        variant="filled"
+        severity="success"
+        action={voteOnSuccessSnackbarAction}
+      >
+        {`Stake succeeded with transaction ${
+          (transactionResponse as TransactionResponseOnSuccess).transactionHash
+        }.`}
+      </Alert>
+    </Snackbar>
+  );
+
+  const voteOnFailureSnackbarAction = (
+    <IconButton
+      size="small"
+      aria-label="close"
+      color="inherit"
+      onClick={onCloseErrorAlert}
+    >
+      <CloseIcon fontSize="small" />
+    </IconButton>
+  );
+
+  const voteOnFailureSnackbar = transactionResponse !== null && (
+    <Snackbar
+      open={!transactionResponse.succeeded}
+      anchorOrigin={{
+        vertical: "top",
+        horizontal: "center",
+      }}
+    >
+      <Alert
+        variant="filled"
+        severity="error"
+        action={voteOnFailureSnackbarAction}
+      >
+        {`Stake failed with error message "${
+          (transactionResponse as TransactionResponseOnFailure).message
+        }". Please try again.`}
+      </Alert>
+    </Snackbar>
+  );
+
   return (
-    <Grid>
-      <Header />
-      <Grid container spacing={2}>
-        <Grid item xs={6} md={6}>
-          <FormControl fullWidth>
-            <InputLabel htmlFor="outlined-adornment-amount">
-              Staking Amount
-            </InputLabel>
-            <OutlinedInput
-              label="Staking Amount"
-              value={stakingAmount}
-              onChange={onStakingAmountChange}
-              startAdornment={
-                <InputAdornment position="start">$</InputAdornment>
-              }
-            />
-          </FormControl>
-        </Grid>
-        <Grid item xs={6} md={6}>
-          <FormControl fullWidth>
-            <LocalizationProvider dateAdapter={AdapterDateFns}>
-              <DateTimePicker
-                label="Staking End Time"
-                value={stakingEndTime}
-                onChange={onStakingEndTimeChange}
-                renderInput={(params) => <TextField {...params} />}
-                minTime={new Date()}
+    <>
+      {voteOnSuccessSnackbar}
+      {voteOnFailureSnackbar}
```

Comment:
Seems like we're reusing the snack bar from the vote buttons. Not a blocker, but let's modularize this part and make it reusable. I can help with this, let me know.

Reply:
agree. was thinking on doing it in a separate PR to keep this one light. feel free to tackle it =)